src/deluge/gui/views/session_view.cpp

```diff
@@ -3303,18 +3317,29 @@ RGB SessionView::gridRenderClipColor(Clip* clip) {
 		}
 	}
 
-	// Black phase of arm flashing
-	if (view.clipArmFlashOn && clip->armState != ArmState::OFF) {
-		return colours::black;
-	}
-
-	// Set a random color if unset and convert to result colour
+	// Set a random colour if unset and convert to result colour
 	if (clip->output->colour == 0) {
 		lastColour = std::fmod(lastColour + colourStep + 192, 192);
 		clip->output->colour = lastColour;
 	}
 
 	RGB resultColour = RGB::fromHue(clip->output->colour);
+	// when selecting a clip, dim all other clip pads a bit
+	// so that the selected clip pad can be lit a bit brighter
+	if (renderPulse && (x != gridFirstPressedX || y != gridFirstPressedY)) {
+		resultColour = resultColour.adjust(255, 2);
+	}
+
+	// Black phase of arm flashing
+	if (view.clipArmFlashOn && clip->armState != ArmState::OFF) {
+		return colours::black;
+	}
+	// If we're currently pulsing this clip and we have the current pulse colour, use that instead
+	else if (!viewingRecordArmingActive && renderPulse && gridSelectedClipPulsing && (clip == selectedClipForPulsing)
```

Comment:
checking gridSelectedClipPulsing is redundant (if it's null then it won't equal clip)

I think checking that the colour isn't black is also redundant - the only time it's the case while pulsing is active is if reset is called, and using the base colour instead will just cause a 50ms blip of the wrong colour